### DISCUSSION TITLE
esbuild docker installation error

### DISCUSSION THREAD

**Discussion by usama-shamim** [chaskiq/chaskiq#883] [Q&A]
Hi, 
Any help will be appreciated. Facing an issue while installing Chaskiq on docker.
using docker compose build && docker compose up to install it.
SO basically all containers start up but rails one is failing with below mentioned error. Not sure how to resolve it. Any help will be appreciated. Just cloned the repo and updated .env file according to the mentioned settings. I am on ubuntu 20.0.4. 

chaskiq-rails-1     | 14:51:37 js.1   | Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'esbuild' imported from /app/esbuild.config.mjs
chaskiq-rails-1     | 14:51:37 js.1   |     at new NodeError (node:internal/errors:387:5)
chaskiq-rails-1     | 14:51:37 js.1   |     at packageResolve (node:internal/modules/esm/resolve:852:9)
chaskiq-rails-1     | 14:51:37 js.1   |     at moduleResolve (node:internal/modules/esm/resolve:901:20)
chaskiq-rails-1     | 14:51:37 js.1   |     at defaultResolve (node:internal/modules/esm/resolve:1115:11)
chaskiq-rails-1     | 14:51:37 js.1   |     at nextResolve (node:internal/modules/esm/loader:163:28)
chaskiq-rails-1     | 14:51:37 js.1   |     at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
chaskiq-rails-1     | 14:51:37 js.1   |     at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
chaskiq-rails-1     | 14:51:37 js.1   |     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
chaskiq-rails-1     | 14:51:37 js.1   |     at link (node:internal/modules/esm/module_job:75:36) {
chaskiq-rails-1     | 14:51:37 js.1   |   code: 'ERR_MODULE_NOT_FOUND'

**Comment by michelson** [comment]
Hi, did you run the command `yarn install` in the rails machine? that should solve the problem. 
please let us know how it goes

**Reply by usama-shamim** [reply]
Hi thanks alot for you reply. 
And yes i ran it. I have used the same Dockerfile.development and Dockerfile  that is provided in the clone.

**Reply by michelson** [reply]
Hi, are you using this for the development environment right?
what is the output of the command when you run the `yarn install`

**Reply by usama-shamim** [reply]
Hi yes its for development environment and i am running docker compose build. Here is the output:

![image](https://user-images.githubusercontent.com/54571862/210322869-2a4964a6-7fbc-4d3a-ad57-94859bc649d2.png)

Here is the content of my Dockerfile.development:

![image](https://user-images.githubusercontent.com/54571862/210323195-9273da4c-16c3-4944-939c-061cab59f562.png)

**Reply by usama-shamim** [reply]
@michelson please confirm if i am missing anything here.

**Reply by michelson** [reply]
Hi, try to run `docker-compose run rails bundle install` and `docker-compose run rails yarn install`
then to run the app use: `docker-compose rails up`

let me know how it goes
